src/http/HttpHeaderBag.php

```diff
@@ -37,28 +59,97 @@ public function get($key) {
     if (array_key_exists($key, $this->values)) {
       return $this->values[$key];
     }
-    return NULL;
+    // Return a NULL object on which you can still call the HttpHeaderInterface
+    // methods.
+    return HttpHeaderNull::create(NULL, NULL);
```

Comment:
This is actually kind of cool (from how I see it). It will allow us doing things like:

```php
if ($value = $headers->get('header-that-does-not-exist')->getValueString()) { … }
```

Instead of the more verbose (and awkard):

```php
if ($header = $headers->has('header-that-does-not-exist')) {
  $value = $headers->get('header-that-does-not-exist')->getValueString();
  …
}
```

https://en.wikipedia.org/wiki/Null_Object_pattern

I don't feel to attached to it if you don't like it, though.